tests/Microsoft.Identity.Web.Test.Integration/AcquireTokenForAppIntegrationTests.cs

```diff
@@ -91,6 +92,7 @@ private void InitializeTokenAcquisitionObjects()
             IOptions<MsalMemoryTokenCacheOptions> tokenOptions = _provider.GetService<IOptions<MsalMemoryTokenCacheOptions>>();
             IOptions<ConfidentialClientApplicationOptions> ccOptions = _provider.GetService<IOptions<ConfidentialClientApplicationOptions>>();
             ILogger<TokenAcquisition> logger = _provider.GetService<ILogger<TokenAcquisition>>();
+            IHttpClientFactory httpClientFactory = _provider.GetService<IHttpClientFactory>();
```

Comment:
I wonder if we can add a better unit test here, as this doesn't really assert anything? At the very we should assert the httpClientFactory is not null. 

Ideally you would inject a mock IHttpClientFactory and make sure that MSAL actually uses it for token acquisition.

Reply:
When you do `BuildTheRequiredServices()` in the test, you do ` services.AddHttpClient();` which adds the httpClient, which will do the DI for the IHttpClientFactory. The implementation of the IHttpClientFactory is tightly tied to the DI implementation, so i'm not sure how we would do this without doing something outside of what actually happens with the library.

Reply:
Fair enough.